globus_automate_client/action_client.py

```diff
@@ -118,54 +112,30 @@ def release(self, action_id: str) -> GlobusHTTPResponse:
         path = self.qjoin_path(action_id, "release")
         return self.post(path)
 
+    @classmethod
+    def new_client(
+        cls, action_url: str, authorizer: GlobusAuthorizer
```

Comment:
I've been mucking about with a few ways to handle these factory things. I've tried:
```
T = TypeVar('T')
@classmethod
def new_thing(cls: Type[T]), ...) -> T:
```

And there's also 
```
from __future__ import annotations
class Foo:
@classmethod
def new_foo(cls, ...) -> Foo:
```

The latter is better and the from future isn't even necessary starting with something like 3.9, but I'm not sure it is even available for < 3.7 (I think we keep 3.6 as a valid version, but I'm not sure we've officially stated that).

So, I guess maybe try the former and especially see if in an editor like VSCode that uses these annotations well it accomplishes what we want in terms of type checking in use and completion and so on.

Reply:
Added this - VSCode seems to reason well enough about the `new_client` return object with the `-> "FlowsClient` notation. I'm kinda impressed we naturally have this variety in editors bc it let's us test what works and doesn't work in a bunch of different editors.

Reply:
yeah, no need to change something that's ok. But, looking forward to not having to do that in the future.